hydrolib/core/basemodel.py

```diff
@@ -441,6 +469,38 @@ def is_empty(self) -> bool:
         """
         return not any(self._cache_dict)
 
+    def exists(self, path: Path) -> bool:
+        """Whether or not the filepath is in the cache.
+
+        Args:
+            path (Path): The path to verify if it is already added in the cache.
+
+        Returns:
+            bool: Whether or not the path is in the cache.
+        """
+        return path in self._cache_dict
+
+    def has_changed(self, path: Path) -> bool:
+        """Whether or not the file in the filepath has changed from the cache.
+
+        Args:
+            path (Path): The path to verify verify against.
+
+        Returns:
+            bool: Whether or the file has changed.
+            True when the file has changed.
+            True when the file does not exist in caching reference.
+            False when the file has not changed.
+        """
+        if not self.exists(path):
+            return True
+
+        checksum = self._get_checksum(path)
+        return checksum != self._cache_dict.get(path).checksum
+
+    def _get_checksum(self, path: Path) -> str:
```

Comment:
If you decide to make `calculate_checksum()` return an optional string, then also make this return value an `Optional[str]`

Reply:
will do